### PR TITLE
Convert tests to XUnit

### PR DESCRIPTION
**Pull request by cstlaurent**
## Goal of the pull request

Convert tests currently running on MSTests to XUnit to remove dependency on Visual Studio. As discussed in issue #62 
## What is needed
- [x] Convert MSTest project to regular class library since the .csproj itself fiddle with MSTest
- [x] Remove MSTest and add XUnit
- [x] Convert all Assert nomenclature to the XUnit one
- [x] Replace all  `if` checks with associated `Assert.Fail` in code since it does not exist in XUnit
- [x] Remove all `TestContext` in code since it does not exist in XUnit
- [x] Validate that all tests are still passing

### CODE REVIEW

**Comment by cstlaurent** [comment]
Hi, I found out a bug in the tests while converting it to XUnit. Actually multiple instance of it...

The bug is in `TestAsyncSubscriptionPending()` inside the `TestSubscriptions` class, line 561 (in the MSTest codebase)

The bug is a testing technique that was used throughout the test code. It's actually a combination of `Assert.Fail()` inside `try-catch` blocks where the Assert.Fail is placed inside the try block after a line of code that should fail. This was to check that the code under test actually throws an exception, as says the Assert message: `"Should have received an exception."`.

The problem is represented in the following screenshot:
image

Here `s.ClearMaxPending();` should throw an exception as intended the test, but it does not, so the `Assert.Fail();` execute, which throws an exception that is swallowed by the empty catch block.

So basically the test was passing not because `s.ClearMaxPending();` was throwing an exception but because the Assert exception was swallowed by a catch block.

And this technique was used all over the place. It's a chance that this is the only failing test after the conversion to XUnit.

**Comment by cstlaurent** [comment]
If I comment out the 3 last Exception checks, the test pass:
- `s.ClearMaxPending()`
- `s.Delivered`
- `s.Dropped`

This is the last thing remaining to finalize this pull-request. Should I leave it this way and have the pull-request accepted with 1 failing test? Or should I comment the 3 failing checks to make the test pass?

**Comment by ColinSullivan1** [comment]
@cstlaurent, I really appreciate the work you've put into this contribution!  Good catch on the tests there - to finalize this you should be able to actually remove the three failing exception tests (those APIs do not throw exceptions - at one point in early development they did).  I'll do some compatibility testing with the project file in the meantime.  Great work!

**Comment by cstlaurent** [comment]
@ColinSullivan1, I removed the invalid checks. I think the pull request is pretty much done on my side at this point.

Thanks!
